### PR TITLE
More informative message at ONNX export

### PR DESCRIPTION
**Pull request by fxmarty**
For example, now warn:

```
The ONNX export succeeded with the warning: The maximum absolute difference between the output of the reference model and the ONNX exported model is not within the set tolerance 1e-05:
- present.7.key: max diff = 1.3113021850585938e-05
- present.11.value: max diff = 1.5974044799804688e-05. The exported model was saved at: gpt2_onnx
```

instead of:

```
An error occured, but the model was saved at: gpt2_onnx
```

Fixes https://github.com/huggingface/optimum/issues/607

### CODE REVIEW

File: optimum/exporters/onnx/__main__.py [hunk 1]
```diff
@@ -194,10 +202,21 @@ def main():
                 atol=args.atol,
             )
 
-    except ValueError:
-        logger.error(f"An error occured, but the model was saved at: {args.output.parent.as_posix()}")
-        return
-    logger.info(f"All good, model saved at: {args.output.parent.as_posix()}")
+        logger.info(f"The ONNX export succeeded and the exported model was saved at: {args.output.parent.as_posix()}")
+    except ShapeError as e:
+        raise ShapeError(e)
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/__main__.py:207]: Quick question, not sure about it:
```suggestion
        raise e
```
?

File: optimum/exporters/onnx/convert.py [hunk 2]
```diff
@@ -40,6 +40,18 @@
...
 
+class ShapeError(ValueError):
+    pass
+
+
+class AtolError(ValueError):
+    pass
+
+
+class OutputError(ValueError):
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/convert.py:51]: A suggestion, feel free to ignore:
```suggestion
class OutputMatchError(ValueError):
```

**Review by michaelbenayoun — APPROVED** [review]
LGTM!

File: optimum/exporters/onnx/__main__.py (hunk 1 above)
Comment (fxmarty) [reply optimum/exporters/onnx/__main__.py]: Thank you!

**Pull request merged**